frontend/src/libs/import-export/normalize-ddah-import.ts

```diff
@@ -0,0 +1,111 @@
+import { Applicant, MinimalDdah } from "../../api/defs/types";
+import {
+    SpreadsheetRowMapper,
+    matchByUtoridOrName,
+} from "../../libs/import-export";
+
+/**
+ * Convert imported spreadsheet or JSON data into an
+ * array of minimal DDAH objects.
+ *
+ * @param {({
+ *     fileType: "json" | "spreadsheet";
+ *     data: any;
+ * })} data
+ * @param {Applicant[]} applicants
+ * @returns {MinimalDdah[]}
+ */
+export function normalizeDdahImports(
+    data: {
+        fileType: "json" | "spreadsheet";
+        data: any;
+    },
+    applicants: Applicant[]
+): MinimalDdah[] {
+    let ret: MinimalDdah[] = [];
+
+    if (data.fileType === "json") {
+        let unwrapped: MinimalDdah[] = data.data;
+        if ((unwrapped as any).ddahs) {
+            unwrapped = (unwrapped as any).ddahs;
+        }
+        for (const ddah of unwrapped) {
+            ret.push(ddah);
+        }
+    }
+
+    if (data.fileType === "spreadsheet") {
+        const unwrapped = data.data;
+        // Get an upper bound for the maximum number of duties that the spreadsheet might have
+        const maxDuties = Math.round(
+            Math.max(
```

Comment:
Why would `...unwrapped.map((row: object) => Object.keys(row).length)` be negative? if not what is the purpose of taking the max between this value and 0?

Reply:
Yes this is weird... But since this code is from original code and not relating to this PR, please make a separate issue and we can look into it.